sonic_platform_base/chassis_base.py

```diff
@@ -0,0 +1,207 @@
+#
+# chassis_base.py
+#
+# Base class for implementing a platform-specific class with which
+# to interact with a chassis device in SONiC.
+#
+
+import sys
+from . import device_base
+
+
+class ChassisBase(device_base.DeviceBase):
+    """
+    Base class for interfacing with a platform chassis
+    """
+
+    # Possible reboot causes
+    REBOOT_CAUSE_POWER_LOSS = "power_loss"
+    REBOOT_CAUSE_THERMAL_OVERLOAD = "thermal_overload"
+    REBOOT_CAUSE_INSUFFICIENT_FAN = "insufficient_fan"
+    REBOOT_CAUSE_WATCHDOG = "watchdog"
+    REBOOT_CAUSE_SOFTWARE = "software"
```

Comment:
Can we differentiate between warm vs cold reboot ?

Reply:
we also need a REBOOT_CAUSE_NA = "na" if hardware cannot determine the exact reason.

Reply:
My intentions were that `REBOOT_CAUSE_SOFTWARE` would be the "N/A", meaning that the reboot was not caused by hardware. I can replace `REBOOT_CAUSE_SOFTWARE` with `REBOOT_CAUSE_NA` if you prefer.

Reply:
Agreed that this is reboot cause and not type.
REBOOT_CAUSE_POWER_LOSS can be changed as REBOOT_CAUSE_POWER_ON_RESET (which is typically the default if FPGA is unable to figure the reason)..

Reply:
@padmanarayana: What do you mean by "FPGA is unable to figure the reason"?

If the hardware is unable to determine a hardware-related reboot cause, the assumption should be that the machine was rebooted by software (i.e., `sudo reboot`). In this case, this method would return `REBOOT_CAUSE_SOFTWARE` (or `REBOOT_CAUSE_NA` if we change the name). I feel this is the scenario you are describing.

On the other hand, `REBOOT_CAUSE_POWER_LOSS` would be returned only if the previous reboot was caused by a hard power loss (i.e., device was unplugged from A/C).